src/gui/windows/multiplayer.zig

```diff
@@ -56,17 +56,28 @@ fn join(_: usize) void {
 		ipAddress = "";
 	}
 	if(connection) |_connection| {
+		if (ipAddressEntry.currentString.items.len == 0) {
+			std.log.err("IP address cannot be empty", .{});
+			main.gui.windowlist.notification.raiseNotification("IP address cannot be empty");
+			return;
+		}
 		_connection.world = &main.game.testWorld;
 		main.globalAllocator.free(settings.lastUsedIPAddress);
 		settings.lastUsedIPAddress = main.globalAllocator.dupe(u8, ipAddressEntry.currentString.items);
 		settings.save();
 		main.game.world = &main.game.testWorld;
+		std.log.info("Connecting to server: {s}", .{ipAddressEntry.currentString.items});
 		main.game.testWorld.init(ipAddressEntry.currentString.items, _connection) catch |err| {
-			std.log.err("Encountered error while opening world: {s}", .{@errorName(err)});
+			const formattedError = std.fmt.allocPrint(main.stackAllocator.allocator, "Encountered error while opening world: {s}", .{@errorName(err)}) catch unreachable;
+			defer main.stackAllocator.free(formattedError);
+			std.log.err("{s}", .{formattedError});
+			main.gui.windowlist.notification.raiseNotification(formattedError);
+			return;
```

Comment:
I just tried to test this, and it seems that this crashes at a later point, before the user has a chance to see your error window.
Could you try to revert all the changes made above, like `main.game.world = &main.game.testWorld;` and `_connection.world = &main.game.testWorld;`, to check if that prevents the crash?

Reply:
that has been happening before i started working on this as far as i can remember. what exactly did you test this with?

another instance of some errors not being caught by the catch (example: unreachable error etc) but logs print the error which led me to believe the issue was with the network checks (try-catch code) somewhere which i deferred to #815.

Reply:
one fix which comes to my mind is:
ip should be in the form of either x.y.z.a or x.y.z.a:b
post this check, all errors are then handled by main.game.testWorld.init's catch part...

Reply:
Are we talking about the same thing?
I just want you to revert the global state changes after printing the error, so we don't have an incorrect world after returning from the function.

Reply:
right. my bad. misunderstood the issue at hand...

Reply:
> Could you try to revert all the changes made above, like main.game.world = &main.game.testWorld; and _connection.world = &main.game.testWorld;, to check if that prevents the crash?

I thought you were referring to the code prior notification gui...

Reply:
✅

Reply:
this resolves #815 as well then...